date_range/models/date_range_search_mixin.py

```diff
@@ -20,9 +25,35 @@ class DateRangeSearchMixin(models.AbstractModel):
         search="_search_date_range_search_id")
 
     def _compute_date_range_search_id(self):
-        """Assign a dummy value for this search field"""
+        """Assign a value for the computed field.
+
+        This can either a matching date range from the specified type,
+        if any, or False.
+        """
+        record2range = {}
+        if self._date_range_assign_type_code:
+            dr_type = self.env["date.range.type"].search(
+                [("code", "=", self._date_range_assign_type_code),
+                 ("company_id", "=", False)], limit=1)
+            if dr_type:
+                query = sql.SQL(
+                    """
+                    SELECT tbl.id, dr.id
+                    FROM {} AS tbl
+                    JOIN date_range AS dr
+                        ON {} BETWEEN dr.date_start AND dr.date_end
+                    WHERE dr.type_id = %s AND tbl.id IN %s
+                    """).format(
+                        sql.Identifier(self._table),
+                        sql.Identifier(self._date_range_search_field),
+                    )
+                self.env.cr.execute(
+                    query,
+                    (dr_type.id, tuple(self.ids or [0])),
+                )
+                record2range = dict(self.env.cr.fetchall())
         for record in self:
-            record.date_range_search_id = False
+            record.date_range_search_id = record2range.get(record.id, False)
```

Comment:
what's the difference between `None` and `False` here?

Reply:
Oh if None works I'll do that.